flang/test/Lower/associate.f90

```diff
@@ -0,0 +1,27 @@
+! RUN: bbc %s -o - | tco | llc --relocation-model=pic --filetype=obj -o %assoc.o
+! RUN: %CC %assoc.o -L%L -lFortranRuntime -lFortranDecimal -lstdc++ -lm -o %assoc.exe
+! RUN: %assoc.exe | FileCheck %s
```

Comment:
I probably did something wrong, but this test failed out of the box.

Reply:
still works for me

Reply:
Looks like a scripting issue.
```
FAIL: Flang :: Lower/associate.f90 (142 of 703)
******************** TEST 'Flang :: Lower/associate.f90' FAILED ********************
Script:
--
: 'RUN: at line 1';   bbc /work/flang/test/Lower/associate.f90 -o - | tco | llc --relocation-model=pic --filetype=obj -o %assoc.o
: 'RUN: at line 2';   /opt/gcc-8.3.0/bin/gcc %assoc.o -L/build/./lib -lFortranRuntime -lFortranDecimal -lstdc++ -lm -o %assoc.exe
: 'RUN: at line 3';   %assoc.exe | /build/bin/FileCheck /work/flang/test/Lower/associate.f90
--
Exit Code: 2

Command Output (stderr):
--
/work/flang/test/Lower/Output/associate.f90.script: line 3: %assoc.exe: command not found
FileCheck error: '<stdin>' is empty.
FileCheck command line:  /build/bin/FileCheck /work/flang/test/Lower/associate.f90

--

********************
********************
Failed Tests (1):
  Flang :: Lower/associate.f90
```

Reply:
It looks like the test (incorrectly) depends on having '.' in `$PATH`.

Are the file names literally `%assoc.o` and `%assoc.exe`? That's a little weird unless there is some convention that applies.

Reply:
Thanks Tim!  It looks like I picked the wrong test to copy from.  Other tests instead use a symbolic temporary file name; chain the RUN commands together; or use ./a.out.  The following works for me.

```
! RUN: bbc %s -o - | tco | llc --relocation-model=pic --filetype=obj -o %t
! RUN: %CC %t -L%L -lFortranRuntime -lFortranDecimal -lstdc++ -lm
! RUN: ./a.out | FileCheck %s
```

Reply:
A new version uses Jean's suggested code.

Reply:
It looks like use of the fixed name _a.out_ can collide with parallel execution of other tests that do that.  Switch to names that start with _%t_, which expands to a full path that is unique to the test.